cocos/2d/CCSpriteFrameCache.cpp

```diff
@@ -391,45 +382,42 @@ void SpriteFrameCache::addSpriteFramesWithFile(const std::string& plist)
         return;
     }
 
-    if (!_spriteFramesCache.hasPlist(plist))
-    {
-        ValueMap dict = FileUtils::getInstance()->getValueMapFromFile(fullPath);
+    ValueMap dict = FileUtils::getInstance()->getValueMapFromFile(fullPath);
 
-        string texturePath("");
+    string texturePath("");
 
-        if (dict.find("metadata") != dict.end())
-        {
-            ValueMap& metadataDict = dict["metadata"].asValueMap();
-            // try to read  texture file name from meta data
-            texturePath = metadataDict["textureFileName"].asString();
-        }
+    if (dict.find("metadata") != dict.end())
+    {
+        ValueMap& metadataDict = dict["metadata"].asValueMap();
+        // try to read  texture file name from meta data
+        texturePath = metadataDict["textureFileName"].asString();
+    }
 
-        if (!texturePath.empty())
-        {
-            // build texture path relative to plist file
-            texturePath = FileUtils::getInstance()->fullPathFromRelativeFile(texturePath, plist);
-        }
-        else
-        {
-            // build texture path by replacing file extension
-            texturePath = plist;
+    if (!texturePath.empty())
+    {
+        // build texture path relative to plist file
+        texturePath = FileUtils::getInstance()->fullPathFromRelativeFile(texturePath, plist);
+    }
+    else
+    {
+        // build texture path by replacing file extension
+        texturePath = plist;
 
-            // remove .xxx
-            size_t startPos = texturePath.find_last_of("."); 
-            texturePath = texturePath.erase(startPos);
+        // remove .xxx
+        size_t startPos = texturePath.find_last_of("."); 
+        texturePath = texturePath.erase(startPos);
 
-            // append .png
-            texturePath = texturePath.append(".png");
+        // append .png
+        texturePath = texturePath.append(".png");
 
-            CCLOG("cocos2d: SpriteFrameCache: Trying to use file %s as texture", texturePath.c_str());
-        }
-        addSpriteFramesWithDictionary(dict, texturePath, plist);
+        CCLOG("cocos2d: SpriteFrameCache: Trying to use file %s as texture", texturePath.c_str());
     }
+    addSpriteFramesWithDictionary(dict, texturePath, plist);
 }
 
 bool SpriteFrameCache::isSpriteFramesWithFileLoaded(const std::string& plist) const
 {
-    return _spriteFramesCache.hasPlist(plist);
+    return _spriteFramesCache.isPlistUsed(plist);
```

Comment:
the meanings of `isSpriteFramesWithFileLoaded ` changed, the origin is
```C++
    /** Check if multiple Sprite Frames from a plist file have been loaded.
    ...
    bool isSpriteFramesWithFileLoaded(const std::string& plist) const;
```
but not it's: Check if __one of__ multiple Sprite Frames from a plist file have been loaded. maybe should change to
```
isPlistUsed && isFullFrames
```